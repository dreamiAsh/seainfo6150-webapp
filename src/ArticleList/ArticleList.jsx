import React from "react";
import styles from "./ArticleList.module.css";
import ArticleListItem from "../ArticleListItem/ArticleListItem";

const ArticleList = (props) => {
    const myList = props.list
    const mappedList = myList.map((listItem) => (
        <li><ArticleListItem article = {listItem}/></li>
    ))
  return (
    <article className={styles.container}>
        <ul className={styles.list}>{mappedList}</ul>
    </article>
  );
};

export default ArticleList;
