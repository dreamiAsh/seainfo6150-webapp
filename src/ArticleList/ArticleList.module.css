.container {
    width: 52%;
    background: #999999;
    border: dashed #6c7075 1px;
    border-radius: 20px;
    font-family: sans-serif;
    padding: 0 15px 0 0;
    box-shadow: 0px 0px 10px red ;
}
.list {
    list-style: none;
}