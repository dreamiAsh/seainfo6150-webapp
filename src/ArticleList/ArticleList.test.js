import React from "react";
import ArticleList from "./ArticleList";
import { fireEvent } from "@testing-library/react";

describe("ArticleList tests", () => {
  it("renders correctly", () => {
    const list = [{"title": "a", 
                    "displayDate": "b", 
                    "shortText": "c", 
                    "image": {
                      "_url": "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
                    }}];
    const { container } = render(<ArticleList list={list}/>);
    expect(container).toMatchSnapshot();
  });
});