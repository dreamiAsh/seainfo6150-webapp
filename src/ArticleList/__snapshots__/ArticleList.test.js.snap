// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <article
    class="container"
  >
    <ul
      class="list"
    >
      <li>
        <article
          class="box"
        >
          <div
            class="imgBox"
          >
            <img
              alt="a"
              class="img"
              src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
            />
          </div>
          <h1
            class="header"
          >
            a
          </h1>
          <div>
            <button
              class="button"
            >
              Show more
            </button>
          </div>
        </article>
      </li>
    </ul>
  </article>
</div>
`;
